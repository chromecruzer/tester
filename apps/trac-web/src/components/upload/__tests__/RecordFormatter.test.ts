import CustomerRecordFormatter from "../CustomerRecordFormatter";

describe('CustomerRecordFormatter', () => {
  let records, modifies;
  beforeEach(() => {
    records = [
      {
        uuid: 'uuid1',
        customer_code: 'customer code',
        name: 'James Tiberius Kirk',
        dea_number: 'DEA 1222',
        hin_number: 'HIM 23333',
        address: '100 Nebraska Way',
        city: 'Metropolis',
        state: 'Nebraska',
        postal: '23245',
        primary_territory: 'Midwest',
        primary_name: 'Larry William Summers',
        key_account_territory: 'Big 8',
        key_account_executive: 'Brian Bedford',
        secondary_territory: 'Arkansa Valley',
        secondary_name: 'Dorothy Bigwind',
        vrs_territory: 'NeverLand',
        vrs_name: 'Peter Pan',
        cos_territory: 'Shangri Pie',
        cos_name: 'Pascal Pythagoras',
        regional_territory: 'Oceana',
        regional_territory_manager_name: 'Tom Sawyer',
        rdb_temp_rdb: 'RDB TEMP I',
        rdb_temp_asd: 'RDB ASD I',
        rdb_temp_1_rdb: 'RDB TEMP 1',
        rdb_temp_1_asd: 'RDB TEMP 2',
        zipcode: '11111',
      },
      {
        uuid: 'uuid2',
        customer_code: 'customer code1',
        name: 'Spock',
        address: '100 Vulcan Path',
        city: 'Pom Far',
        state: 'Starlight',
        postal: '33333',
        primary_territory: 'EAST',
        primary_name: 'Betty Winters',
        key_account_territory: 'Cluster 15',
        key_account_executive: 'Brianna Tablecrossing',
        secondary_territory: 'Starview Valley',
        secondary_name: 'Don LittleBreeze',
        vrs_territory: 'SometimesCountry',
        vrs_name: 'Paul Piglet',
        cos_territory: 'Hong Fang',
        cos_name: 'Julius Shakespeare',
        regional_territory: 'Globe',
        regional_territory_manager_name: 'Huck Finn',
        rdb_temp_rdb: 'RDB TEMP II',
        rdb_temp_asd: 'RDB ASD II',
        rdb_temp_1_rdb: 'RDB TEMP 3',
        rdb_temp_1_asd: 'RDB TEMP 4',
        zipcode: '77777',
      },
    ];
    modifies = [
      {
        original: {
          uuid: 'uuid1',
          customer_code: 'customer code',
          name: 'James Tiberius Kirk',
          dea_number: 'DEA 1222',
          hin_number: 'HIM 23333',
          address: '100 Nebraska Way',
          city: 'Metropolis',
          state: 'Nebraska',
          postal: '23245',
          primary_territory: 'Midwest',
          primary_name: 'Larry William Summers',
          key_account_territory: 'Big 8',
          key_account_executive: 'Brian Bedford',
          secondary_territory: 'Arkansa Valley',
          secondary_name: 'Dorothy Bigwind',
          vrs_territory: 'NeverLand',
          vrs_name: 'Peter Pan',
          cos_territory: 'Shangri Pie',
          cos_name: 'Pascal Pythagoras',
          regional_territory: 'Oceana',
          regional_territory_manager_name: 'Tom Sawyer',
          rdb_temp_rdb: 'RDB TEMP I',
          rdb_temp_asd: 'RDB ASD I',
          rdb_temp_1_rdb: 'RDB TEMP 1',
          rdb_temp_1_asd: 'RDB TEMP 2',
          zipcode: '11111',
        },
        update: {
          uuid: 'uuid1',
          customer_code: 'customer code',
          name: 'James Tiberius Kirk',
          dea_number: 'DEA 1222',
          hin_number: 'HIM 23333',
          address: '100 Nebraska Way',
          city: 'Metropolis',
          state: 'Nebraska',
          postal: '23245',
          primary_territory: 'Midwest',
          primary_name: 'Larry Wally Summers',
          key_account_territory: 'Big 8',
          key_account_executive: 'Brian Bedford',
          secondary_territory: 'Arkansa Valley',
          secondary_name: 'Dorothy Bigwind',
          vrs_territory: 'NeverLand',
          vrs_name: 'Peter Pan',
          cos_territory: 'Shanghi Pie',
          cos_name: 'Pascal Pythagoras',
          regional_territory: 'Oceana',
          regional_territory_manager_name: 'Tom Sawyer',
          rdb_temp_rdb: 'RDB TEMP I',
          rdb_temp_asd: 'RDB ASD I',
          rdb_temp_1_rdb: 'RDB TEMP 1',
          rdb_temp_1_asd: 'RDB TEMP 2',
          zipcode: '11311',
        },
      },
      {
        original: {
          uuid: 'uuid2',
          customer_code: 'customer code1',
          name: 'Spock',
          address: '100 Vulcan Path',
          city: 'Pom Far',
          state: 'Starlight',
          postal: '33333',
          primary_territory: 'EAST',
          primary_name: 'Betty Winters',
          key_account_territory: 'Cluster 15',
          key_account_executive: 'Brianna Tablecrossing',
          secondary_territory: 'Starview Valley',
          secondary_name: 'Don LittleBreeze',
          vrs_territory: 'SometimesCountry',
          vrs_name: 'Paul Piglet',
          cos_territory: 'Hong Fang',
          cos_name: 'Julius Shakespeare',
          regional_territory: 'Globe',
          regional_territory_manager_name: 'Huck Finn',
          rdb_temp_rdb: 'RDB TEMP II',
          rdb_temp_asd: 'RDB ASD II',
          rdb_temp_1_rdb: 'RDB TEMP 3',
          rdb_temp_1_asd: 'RDB TEMP 4',
          zipcode: '77777',
        },
        update: {
          uuid: 'uuid2',
          customer_code: 'customer code1',
          name: 'Worf',
          address: '100 Kapla Path',
          city: 'Pom Far',
          state: 'Starlight',
          postal: '33333',
          primary_territory: 'EAST',
          primary_name: 'Betty Winters',
          key_account_territory: 'Cluster 15',
          key_account_executive: 'Brianna Tablecrossing',
          secondary_territory: 'Starview Valley',
          secondary_name: 'Don LittleBreeze',
          vrs_territory: 'SometimesCountry',
          vrs_name: 'Paul Sabertooth',
          cos_territory: 'Hong Fang',
          cos_name: 'Julius Shakespeare',
          regional_territory: 'Globe',
          regional_territory_manager_name: 'Huck Finn',
          rdb_temp_rdb: 'RDB TEMP II',
          rdb_temp_asd: 'RDB ASD II',
          rdb_temp_1_rdb: 'RDB TEMP 3',
          rdb_temp_1_asd: 'RDB TEMP 4',
          zipcode: '77777',
        },
      },
    ];
  });
  describe('#recordsToTableData', () => {
    it('should convert add comparisons to a list of table records correctly', () => {
      const formatter = new CustomerRecordFormatter({
        adds: records,
        removes: [],
        modifies: [],
      });
      expect(formatter.formattedRecords).toMatchSnapshot();
    });
    it('should convert remove comparisons to a list of table records correctly', () => {
      const formatter = new CustomerRecordFormatter({
        adds: [],
        removes: records,
        modifies: [],
      });
      expect(formatter.formattedRecords).toMatchSnapshot();

    });
    it('should convert modify comparisons to a list of table records correctly', () => {
      const formatter = new CustomerRecordFormatter({
        adds: records,
        removes: [],
        modifies,
      });
      expect(formatter.formattedRecords).toMatchSnapshot();
    });
  });
});
