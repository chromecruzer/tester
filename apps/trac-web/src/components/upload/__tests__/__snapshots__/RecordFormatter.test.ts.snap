// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerRecordFormatter #recordsToTableData should convert add comparisons to a list of table records correctly 1`] = `
Array [
  Object {
    "action": "Added",
    "address": "100 Nebraska Way",
    "city": "Metropolis",
    "cos_name": "Pascal Pythagoras",
    "cos_territory": "Shangri Pie",
    "customer_code": "customer code",
    "dea_number": "DEA 1222",
    "hin_number": "HIM 23333",
    "key_account_executive": "Brian Bedford",
    "key_account_territory": "Big 8",
    "name": "James Tiberius Kirk",
    "postal": "23245",
    "primary_name": "Larry William Summers",
    "primary_territory": "Midwest",
    "rdb_temp_1_asd": "RDB TEMP 2",
    "rdb_temp_1_rdb": "RDB TEMP 1",
    "rdb_temp_asd": "RDB ASD I",
    "rdb_temp_rdb": "RDB TEMP I",
    "regional_territory": "Oceana",
    "regional_territory_manager_name": "Tom Sawyer",
    "secondary_name": "Dorothy Bigwind",
    "secondary_territory": "Arkansa Valley",
    "state": "Nebraska",
    "uuid": "uuid1",
    "vrs_name": "Peter Pan",
    "vrs_territory": "NeverLand",
    "zipcode": "11111",
  },
  Object {
    "action": "Added",
    "address": "100 Vulcan Path",
    "city": "Pom Far",
    "cos_name": "Julius Shakespeare",
    "cos_territory": "Hong Fang",
    "customer_code": "customer code1",
    "key_account_executive": "Brianna Tablecrossing",
    "key_account_territory": "Cluster 15",
    "name": "Spock",
    "postal": "33333",
    "primary_name": "Betty Winters",
    "primary_territory": "EAST",
    "rdb_temp_1_asd": "RDB TEMP 4",
    "rdb_temp_1_rdb": "RDB TEMP 3",
    "rdb_temp_asd": "RDB ASD II",
    "rdb_temp_rdb": "RDB TEMP II",
    "regional_territory": "Globe",
    "regional_territory_manager_name": "Huck Finn",
    "secondary_name": "Don LittleBreeze",
    "secondary_territory": "Starview Valley",
    "state": "Starlight",
    "uuid": "uuid2",
    "vrs_name": "Paul Piglet",
    "vrs_territory": "SometimesCountry",
    "zipcode": "77777",
  },
]
`;

exports[`CustomerRecordFormatter #recordsToTableData should convert modify comparisons to a list of table records correctly 1`] = `
Array [
  Object {
    "action": "Added",
    "address": "100 Nebraska Way",
    "city": "Metropolis",
    "cos_name": "Pascal Pythagoras",
    "cos_territory": "Shangri Pie",
    "customer_code": "customer code",
    "dea_number": "DEA 1222",
    "hin_number": "HIM 23333",
    "key_account_executive": "Brian Bedford",
    "key_account_territory": "Big 8",
    "name": "James Tiberius Kirk",
    "postal": "23245",
    "primary_name": "Larry William Summers",
    "primary_territory": "Midwest",
    "rdb_temp_1_asd": "RDB TEMP 2",
    "rdb_temp_1_rdb": "RDB TEMP 1",
    "rdb_temp_asd": "RDB ASD I",
    "rdb_temp_rdb": "RDB TEMP I",
    "regional_territory": "Oceana",
    "regional_territory_manager_name": "Tom Sawyer",
    "secondary_name": "Dorothy Bigwind",
    "secondary_territory": "Arkansa Valley",
    "state": "Nebraska",
    "uuid": "uuid1",
    "vrs_name": "Peter Pan",
    "vrs_territory": "NeverLand",
    "zipcode": "11111",
  },
  Object {
    "action": "Added",
    "address": "100 Vulcan Path",
    "city": "Pom Far",
    "cos_name": "Julius Shakespeare",
    "cos_territory": "Hong Fang",
    "customer_code": "customer code1",
    "key_account_executive": "Brianna Tablecrossing",
    "key_account_territory": "Cluster 15",
    "name": "Spock",
    "postal": "33333",
    "primary_name": "Betty Winters",
    "primary_territory": "EAST",
    "rdb_temp_1_asd": "RDB TEMP 4",
    "rdb_temp_1_rdb": "RDB TEMP 3",
    "rdb_temp_asd": "RDB ASD II",
    "rdb_temp_rdb": "RDB TEMP II",
    "regional_territory": "Globe",
    "regional_territory_manager_name": "Huck Finn",
    "secondary_name": "Don LittleBreeze",
    "secondary_territory": "Starview Valley",
    "state": "Starlight",
    "uuid": "uuid2",
    "vrs_name": "Paul Piglet",
    "vrs_territory": "SometimesCountry",
    "zipcode": "77777",
  },
  Object {
    "action": "Modified",
    "address": "100 Nebraska Way",
    "city": "Metropolis",
    "cos_name": "Pascal Pythagoras",
    "cos_territory": "Shanghi Pie",
    "cos_territory_original": "Shangri Pie",
    "customer_code": "customer code",
    "dea_number": "DEA 1222",
    "hin_number": "HIM 23333",
    "key_account_executive": "Brian Bedford",
    "key_account_territory": "Big 8",
    "name": "James Tiberius Kirk",
    "postal": "23245",
    "primary_name": "Larry Wally Summers",
    "primary_name_original": "Larry William Summers",
    "primary_territory": "Midwest",
    "rdb_temp_1_asd": "RDB TEMP 2",
    "rdb_temp_1_rdb": "RDB TEMP 1",
    "rdb_temp_asd": "RDB ASD I",
    "rdb_temp_rdb": "RDB TEMP I",
    "regional_territory": "Oceana",
    "regional_territory_manager_name": "Tom Sawyer",
    "secondary_name": "Dorothy Bigwind",
    "secondary_territory": "Arkansa Valley",
    "state": "Nebraska",
    "uuid": "uuid1",
    "vrs_name": "Peter Pan",
    "vrs_territory": "NeverLand",
    "zipcode": "11311",
    "zipcode_original": "11111",
  },
  Object {
    "action": "Modified",
    "address": "100 Kapla Path",
    "address_original": "100 Vulcan Path",
    "city": "Pom Far",
    "cos_name": "Julius Shakespeare",
    "cos_territory": "Hong Fang",
    "customer_code": "customer code1",
    "key_account_executive": "Brianna Tablecrossing",
    "key_account_territory": "Cluster 15",
    "name": "Worf",
    "name_original": "Spock",
    "postal": "33333",
    "primary_name": "Betty Winters",
    "primary_territory": "EAST",
    "rdb_temp_1_asd": "RDB TEMP 4",
    "rdb_temp_1_rdb": "RDB TEMP 3",
    "rdb_temp_asd": "RDB ASD II",
    "rdb_temp_rdb": "RDB TEMP II",
    "regional_territory": "Globe",
    "regional_territory_manager_name": "Huck Finn",
    "secondary_name": "Don LittleBreeze",
    "secondary_territory": "Starview Valley",
    "state": "Starlight",
    "uuid": "uuid2",
    "vrs_name": "Paul Sabertooth",
    "vrs_name_original": "Paul Piglet",
    "vrs_territory": "SometimesCountry",
    "zipcode": "77777",
  },
]
`;

exports[`CustomerRecordFormatter #recordsToTableData should convert remove comparisons to a list of table records correctly 1`] = `
Array [
  Object {
    "action": "Removed",
    "address": "100 Nebraska Way",
    "city": "Metropolis",
    "cos_name": "Pascal Pythagoras",
    "cos_territory": "Shangri Pie",
    "customer_code": "customer code",
    "dea_number": "DEA 1222",
    "hin_number": "HIM 23333",
    "key_account_executive": "Brian Bedford",
    "key_account_territory": "Big 8",
    "name": "James Tiberius Kirk",
    "postal": "23245",
    "primary_name": "Larry William Summers",
    "primary_territory": "Midwest",
    "rdb_temp_1_asd": "RDB TEMP 2",
    "rdb_temp_1_rdb": "RDB TEMP 1",
    "rdb_temp_asd": "RDB ASD I",
    "rdb_temp_rdb": "RDB TEMP I",
    "regional_territory": "Oceana",
    "regional_territory_manager_name": "Tom Sawyer",
    "secondary_name": "Dorothy Bigwind",
    "secondary_territory": "Arkansa Valley",
    "state": "Nebraska",
    "uuid": "uuid1",
    "vrs_name": "Peter Pan",
    "vrs_territory": "NeverLand",
    "zipcode": "11111",
  },
  Object {
    "action": "Removed",
    "address": "100 Vulcan Path",
    "city": "Pom Far",
    "cos_name": "Julius Shakespeare",
    "cos_territory": "Hong Fang",
    "customer_code": "customer code1",
    "key_account_executive": "Brianna Tablecrossing",
    "key_account_territory": "Cluster 15",
    "name": "Spock",
    "postal": "33333",
    "primary_name": "Betty Winters",
    "primary_territory": "EAST",
    "rdb_temp_1_asd": "RDB TEMP 4",
    "rdb_temp_1_rdb": "RDB TEMP 3",
    "rdb_temp_asd": "RDB ASD II",
    "rdb_temp_rdb": "RDB TEMP II",
    "regional_territory": "Globe",
    "regional_territory_manager_name": "Huck Finn",
    "secondary_name": "Don LittleBreeze",
    "secondary_territory": "Starview Valley",
    "state": "Starlight",
    "uuid": "uuid2",
    "vrs_name": "Paul Piglet",
    "vrs_territory": "SometimesCountry",
    "zipcode": "77777",
  },
]
`;
